### PR TITLE
Import CGFloat as CoreFoundation.CGFloat if possible

### PR DESCRIPTION
**Pull request by salinas-miguel**
Resolves rdar://87605416

### CODE REVIEW

File: lib/ClangImporter/ImportDecl.cpp [hunk 1]
```diff
@@ -250,6 +250,29 @@ getSwiftStdlibType(const clang::TypedefNameDecl *D,
     }
 #include "MappedTypes.def"
 
+    if (Name.str() == "CGFloat") {
+      SwiftModuleName = StringRef("CoreGraphics");
```
Comment (slavapestov) [inline lib/ClangImporter/ImportDecl.cpp:254]: I'm not sure the StringRef()s are necessary

**Comment by CodaFi** [comment]
@swift-ci test source compatibility

**Comment by salinas-miguel** [comment]
Multiple source failures producing this error: `error: emit-module command failed due to signal 6 (use -v to see invocation)Assertion failed: (CMU && CMU->isTopLevel() && "Only top-level modules supported"), function FilteringDeclaredDeclConsumer, file ClangImporter.cpp, line 2767.`
I don't understand what's failing here. Is there any way to to see which source file fails to compile now?

**Comment by salinas-miguel** [comment]
Also, is there a way to run this test suite locally?

**Comment by nkcsgexi** [comment]
Also adding @xymus as an additional reviewer. Do we really need to hard-code this information into the importer? Does it imply a deeper issue in modulization.

**Comment by beccadax** [comment]
@swift-ci please smoke test

**Comment by beccadax** [comment]
@swift-ci please test source compatibility

**Comment by airspeedswift** [comment]
@swift-ci please smoke test

File: lib/ClangImporter/ImportDecl.cpp [hunk 2]
```diff
@@ -437,16 +437,20 @@ getSwiftStdlibType(const clang::TypedefNameDecl *D,
...
 
   Type SwiftType = Impl.getNamedSwiftType(M, SwiftTypeName);
+
+  if (!SwiftType && CTypeKind == MappedCTypeKind::CGFloat) {
+    // Fall back to looking for CGFloat in Core Graphics.
+    M = Impl.getNamedModule("CoreGraphics");
+    SwiftType = Impl.getNamedSwiftType(M, SwiftTypeName);
+  }
+
   if (!SwiftType && !CanBeMissing) {
```
Comment (DougGregor) [inline lib/ClangImporter/ImportDecl.cpp:449]: There's a slight behavior change here that might explain the failure in `module-cache-diagnostics.swift`: when there is no module and `CanBeMissing`, the old code would unconditionally return `std::make_pair(Type(), "")`. The new code will return `std::make_pair(Type(), SwiftTypeName)`.

**Review by DougGregor — APPROVED** [review]
This looks okay if it's passing tests

File: lib/ClangImporter/ImportDecl.cpp (hunk 2 above)
Comment (salinas-miguel) [reply lib/ClangImporter/ImportDecl.cpp:449]: Thanks for catching that. I'll add the module check back

**Comment by salinas-miguel** [comment]
The lldb failure was due to the fact that lldb reuses the ClangImporter/MappedTypes.def file. Inverting the fallback logic and keeping the MappedTypes.def file unmodified fixes the failure. This should pass the smoke test now. Can someone please run the smoke test?

**Comment by airspeedswift** [comment]
@swift-ci please smoke test

**Comment by salinas-miguel** [comment]
The Windows failure looks unrelated. #40954 is also hitting the same failure.

**Comment by salinas-miguel** [comment]
@swift-ci please test Windows

**Pull request merged**